File: localstorage/src/main/java/ch/fhnw/geiger/localstorage/db/data/Field.java
package ch.fhnw.geiger.localstorage.db.data;

import ch.fhnw.geiger.localstorage.SearchCriteria;

/**
 * <p>Field reflects the available ordinals in a node.</p>
 */
public enum Field {
  OWNER(1, SearchCriteria.ComparatorType.STRING),
  NAME(2, SearchCriteria.ComparatorType.STRING),
  PATH(3, SearchCriteria.ComparatorType.STRING),
  KEY(4, SearchCriteria.ComparatorType.STRING),
  VALUE(5, SearchCriteria.ComparatorType.STRING),
  TYPE(6, SearchCriteria.ComparatorType.STRING),
  VISIBILITY(7, SearchCriteria.ComparatorType.STRING),
  LAST_MODIFIED(8, SearchCriteria.ComparatorType.DATETIME);

  private final SearchCriteria.ComparatorType comparator;
  private final int id;

  Field(int id, SearchCriteria.ComparatorType comparator) {
    this.comparator = comparator;
    this.id = id;
  }

  public SearchCriteria.ComparatorType getComparator() {
    return comparator;
  }

  public int getId() {
    return id;
  }

}

File: lib/src/db/data/field.dart
library geiger_localstorage;

import 'package:geiger_localstorage/geiger_localstorage.dart';

/// <p>Field reflects the available ordinals in a node.</p>
class Field {
  static const Field owner = Field._(1, ComparatorType.string);
  static const Field name = Field._(2, ComparatorType.string);
  static const Field path = Field._(3, ComparatorType.string);
  static const Field key = Field._(4, ComparatorType.string);
  static const Field value = Field._(5, ComparatorType.string);
  static const Field type = Field._(6, ComparatorType.string);
  static const Field visibility = Field._(7, ComparatorType.string);
  static const Field lastModified = Field._(8, ComparatorType.datetime);
  static const Field expiry = Field._(9, ComparatorType.datetime);
  static const Field tombstone = Field._(10, ComparatorType.boolean);

  final ComparatorType _comparator;

  ComparatorType get comparator => _comparator;

  final int _id;

  int get id => _id;

  const Field._(this._id, this._comparator);
}
